### PR TITLE
Add ability to change users's passwords

### PR DESCRIPTION
**Pull request by louis-ver**

### CODE REVIEW

File: server/common/error_codes_and_messages.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def errors_as_list() -> List:
     35, "'property' must be specified to use the 'propertyValue' argument")
 USERNAME_ALREADY_EXISTS = ErrorCodeAndMessage(40,
                                               "Username '{}' already exists")
+USER_DOES_NOT_EXIST = ErrorCodeAndMessage(41, "Username '{}' does not exist")
```
Comment (simon-dube) [inline server/common/error_codes_and_messages.py:50]: At first I wasn't sure about this error message as this provides a way to users to find out what username already exists or do not exist, but as it is only used for when an administrator tries to edit a user's password, it's correct as it would require the attacker to be an admin.

File: server/resources/edit.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                # Logged in user is not an admin
+                if model.username and model.username != user.username:
+                    return UNAUTHORIZED
+                edit_user = db.session.query(User).filter_by(
+                    username=user.username).first()
+            edit_user.password = generate_password_hash(model.password)
+            db.session.commit()
+        else:
+            # Password was not provided
+            return INVALID_MODEL_PROVIDED
```
Comment (simon-dube) [inline server/resources/edit.py]: Could you please add error_detail saying "missing password"? This is what the `unmarshal_request` would normally do.

File: server/test/resources/test_edit.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class TestEditResource():
+    def test_edit_password_admin(self, test_client):
+        response = test_client.post(
+            "/users/edit",
+            headers={"apiKey": admin().api_key},
+            data=json.dumps({
+                "username": standard_user().username,
+                "password": "new_password"
```
Comment (simon-dube) [inline server/test/resources/test_edit.py]: I usually prefer using the default user password + something to insure this was never added to the DB in case we someday change the standard_user password to "new_password".

File: server/test/resources/test_edit.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    def test_edit_password_no_password(self, test_client):
+        response = test_client.post(
+            "/users/edit",
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps({
+                "password": ""
+            }))
+        error = error_from_response(response)
+        assert error == INVALID_MODEL_PROVIDED
```
Comment (simon-dube) [inline server/test/resources/test_edit.py]: Please add a test (maybe part of `test_edit_password_no_password`) where the json dumps is simply empty.

**Review by simon-dube — APPROVED** [review]
Few possible changes. Other than that, great job.

File: server/test/resources/test_edit.py (hunk 4 above)
Comment (louis-ver) [reply server/test/resources/test_edit.py]: I thought of doing this, but this situation is handled by the decorator, not the function itself. That is why I have the test for `{"password": ""}`.
